src/utils/who.is.making.filing.ts

```diff
@@ -21,18 +26,41 @@ export const getWhoIsFiling = (req: Request, res: Response, next: NextFunction,
   }
 };
 
-export const postWhoIsFiling = (req: Request, res: Response, next: NextFunction, agentUrl: string, oeUrl: string) => {
+export const postWhoIsFiling = async (
+  req: Request,
+  res: Response,
+  next: NextFunction,
+  agentUrl: string,
+  oeUrl: string,
+  isRegistrationJourney: boolean = false
+): Promise<void> => {
+
   try {
+
     logger.debugRequest(req, `${req.method} ${req.route.path}`);
+
+    const appData: ApplicationData = getApplicationData(req.session);
     const whoIsRegistering = req.body[WhoIsRegisteringKey];
+    appData[WhoIsRegisteringKey] = whoIsRegistering;
+    const session = req.session as Session;
 
-    setExtraData(req.session, { ...getApplicationData(req.session), [WhoIsRegisteringKey]: whoIsRegistering });
+    let nextPageUrl: string = "";
 
-    if (whoIsRegistering === WhoIsRegisteringType.AGENT){
-      return res.redirect(agentUrl);
-    } else {
-      return res.redirect(oeUrl);
+    if (whoIsRegistering === WhoIsRegisteringType.AGENT) {
+      nextPageUrl = agentUrl;
+    }
+    if (whoIsRegistering === WhoIsRegisteringType.SOMEONE_ELSE) {
+      nextPageUrl = oeUrl;
+    }
+    if (isActiveFeature(config.FEATURE_FLAG_ENABLE_REDIS_REMOVAL) && isRegistrationJourney) {
+      if (appData[Transactionkey] && appData[OverseasEntityKey]) {
+        await updateOverseasEntity(req, session, appData);
+      } else {
+        throw new Error("Error: who_is_registering filter cannot be updated - transaction_id or overseas_entity_id is missing");
```

Comment:
i would have 'page' in the message after 'filter' 
eg
Error: who_is_registering filter page cannot ...